services/appflowy-collaborate/src/indexer/document_indexer.rs

```diff
@@ -42,16 +45,42 @@ impl Indexer for DocumentIndexer {
     match result {
       Ok(document_data) => {
         let content = document_data.to_plain_text();
-        let plain_text_param = AFCollabEmbeddingParams {
-          fragment_id: object_id.clone(),
-          object_id: object_id.clone(),
-          collab_type: CollabType::Document,
-          content_type: EmbeddingContentType::PlainText,
-          content,
-          embedding: None,
+        let mut result = Vec::with_capacity(1 + content.len() / Self::DOC_CONTENT_SPLIT);
+
+        let mut slice = content.as_str();
+        while slice.len() > Self::DOC_CONTENT_SPLIT {
```

Comment:
I can handle content splitting directly in the AppFlowy-AI service, eliminating the need to split it in AppFlowy Cloud using DOC_CONTENT_SPLIT.

Reply:
From what I've noticed, text splitting in AppFlowy-AI service is not what we need: It uses either \n\n for text splitting (which roughly means splitting by paragraph) or splitting by code formatting scopes. Neither of these is good for our use case - we were explicitly splitting by block in the past and it turned out to be too expensive, which is why we are at the current state.